Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -498,6 +494,77 @@ class TabTrayController: UIViewController {
         mvc.fixedWidth = TabTrayControllerUX.MenuFixedWidth
         self.presentViewController(mvc, animated: true, completion: nil)
     }
+
+    @available(iOS 9, *)
+    func SELdidTapMask() {
+        attemptToTogglePrivateMode()
+    }
+
+    @available(iOS 9, *)
+    func attemptToTogglePrivateMode() -> Success {
+        guard let navigationController = self.navigationController else {
+            return deferMaybe(AuthorisationError(description: "Failed to switch the private mode due to an inexistent navigation controller."))
+        }
+        let success = Success()
+        tabManager.authorisePrivateMode(navigationController).uponQueue(dispatch_get_main_queue()) { result in
+            if let _ = result.successValue {
```

Comment:
Nit: result has a `isSuccess` boolean method you can use instead of assigning to a blank.